quickwit-config/src/config.rs

```diff
@@ -167,6 +167,7 @@ pub struct QuickwitConfig {
     #[serde(default = "default_metastore_and_index_root_uri")]
     pub default_index_root_uri: String,
     #[serde(default = "default_data_dir_path")]
+    #[serde(rename = "data_dir")]
```

Comment:
I thought we were already using `data_dir` in the conf but it wasn't. I like keeping the short name for the config and the long name in the code. It's also in line with the env variable `QW_DATA_DIR`.